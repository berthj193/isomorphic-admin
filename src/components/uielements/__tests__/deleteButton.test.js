import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import _omit from 'lodash/omit';

import DeleteButton from '../deleteButton';

describe('<DeleteButton />', () => {
  const props = {
    onConfirm: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should render icon on default', () => {
    expect(toJson(shallow(<DeleteButton {...props} />))).toMatchSnapshot();
  });

  it('should render children, if they were passed', () => {
    expect(toJson(shallow(<DeleteButton {...props}><a>Link!</a></DeleteButton>))).toMatchSnapshot();
  });

  it('should pass all props, besides tooltipText and children, to AntD component', () => {
    const customProps = {
      onConfirm: jest.fn(),
      onCancel: jest.fn(),
      okText: 'Thats ok!',
      cancelText: 'That\'s not okay:(',
      title: 'Are you REALLY sure?',
      children: <span />,
      placement: 'top',
      tooltipText: 'You touched me!',
    };

    const popconfirmInstance = mount(<DeleteButton {...customProps} />)
      .find('Popconfirm')
      .first()
      .instance();

    expect(popconfirmInstance.props)
      .toMatchObject(_omit(customProps, 'tooltipText', 'onCancel', 'children'));
  });

  describe('preventEvent', () => {
    const mockEvent = {
      stopPropagation: jest.fn(),
      preventDefault: jest.fn(),
    };

    it('should prevent event from propagating', () => {
      DeleteButton.preventEvent(mockEvent);
      expect(mockEvent.stopPropagation).toHaveBeenCalled();
    });

    it('should prevent defaultm behaviour', () => {
      DeleteButton.preventEvent(mockEvent);
      expect(mockEvent.preventDefault).toHaveBeenCalled();
    });
  });
});
