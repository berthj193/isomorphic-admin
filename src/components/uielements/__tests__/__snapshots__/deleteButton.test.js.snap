// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteButton /> should render children, if they were passed 1`] = `
<Popconfirm
  cancelText="Cancel"
  okText="Delete"
  okType="primary"
  onCancel={[Function]}
  onConfirm={[Function]}
  placement="bottom"
  prefixCls="ant-popover"
  title="Are you sure?"
  transitionName="zoom-big"
  trigger="click"
>
  <a>
    Link!
  </a>
</Popconfirm>
`;

exports[`<DeleteButton /> should render icon on default 1`] = `
<Popconfirm
  cancelText="Cancel"
  okText="Delete"
  okType="primary"
  onCancel={[Function]}
  onConfirm={[Function]}
  placement="bottom"
  prefixCls="ant-popover"
  title="Are you sure?"
  transitionName="zoom-big"
  trigger="click"
>
  <a
    onClick={[Function]}
  >
    <Tooltip
      arrowPointAtCenter={false}
      autoAdjustOverflow={true}
      mouseEnterDelay={0.1}
      mouseLeaveDelay={0.1}
      placement="top"
      prefixCls="ant-tooltip"
      title="Remove"
      transitionName="zoom-big-fast"
    >
      <Icon
        type="delete"
      />
    </Tooltip>
  </a>
</Popconfirm>
`;
